src/features/productDetails/ProductDetails.tsx

```diff
@@ -5,19 +5,27 @@ import {getProductWithId} from "../../services/api.ts";
 import {useCartStore} from "../cart/cartStore.ts";
 import ProductDetailsLoader from "./ProductDetailsLoader.tsx";
 import AppError from "../error/AppError.tsx";
+import {useWishlistStore} from "../wishlist/wishlistStore.ts";
+
 
 const ProductDetails = () => {
-    const {productId} = useParams()
-    const productQuery = useQuery({queryKey: [productId], queryFn: () => getProductWithId(productId ?? '')})
+    const {productId} = useParams();
+    const productQuery = useQuery({queryKey: [productId], queryFn: () => getProductWithId(productId ?? '')});
+
     const addItemToCart = useCartStore((state) => state.addItem);
+    const {wishlist, addToWishlist, removeFromWishlist} = useWishlistStore((state) => ({
```

Comment:
You can just use: 

const {wishlist, addToWishlist, removeFromWishlist} = useWishlistStore();

No need to manually assign them(since you are using same naming conventions from your store) :)

Reply:
and similar in the other places your have set them :)